From: Orit Wasserman <owasserm@redhat.com>
Date: Tue, 11 Oct 2016 11:19:01 +0200
Subject: rgw: Fix missing master zone for default region conversion

Fixes:http://tracker.ceph.com/issues/17371
Signed-off-by: Orit Wasserman <owasserm@redhat.com>
(cherry picked from commit cd3bccdef2ecd1d294ab1192ab3b9e731b5592e0)

Resolves: rhbz#1383631
---
 src/rgw/rgw_rados.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/rgw/rgw_rados.cc b/src/rgw/rgw_rados.cc
index f77f07a..360e722 100644
--- a/src/rgw/rgw_rados.cc
+++ b/src/rgw/rgw_rados.cc
@@ -3426,6 +3426,11 @@ int RGWRados::replace_region_with_zonegroup()
       return ret;
     }
     zonegroup.realm_id = realm.get_id();
+    /* fix default region master zone */
+    if (*iter == default_zonegroup_name && zonegroup.master_zone.empty()) {
+      ldout(cct, 0) << "Setting default zone as master for default region" << dendl;
+      zonegroup.master_zone = default_zone_name;
+    }
     ret = zonegroup.update();
     if (ret < 0 && ret != -EEXIST) {
       ldout(cct, 0) << "failed to update zonegroup " << *iter << ": ret "<< ret << " " << cpp_strerror(-ret)
